src/NuGet.Core/NuGet.Configuration/Settings/Items/SourceItem.cs

```diff
@@ -62,23 +46,5 @@ public override SettingBase Clone()
 
             return newSetting;
         }
-
-        public override bool Equals(object other)
-        {
-            var source = other as SourceItem;
-
-            if (source == null)
-            {
-                return false;
-            }
-
-            if (ReferenceEquals(this, source))
-            {
-                return true;
-            }
-
-            return string.Equals(Key, source.Key, StringComparison.Ordinal) &&
-                string.Equals(ProtocolVersion, source.ProtocolVersion, StringComparison.Ordinal);
-        }
```

Comment:
There are a few [tests ](https://github.com/NuGet/NuGet.Client/blob/dev/test/NuGet.Core.Tests/NuGet.Configuration.Test/SettingsFileParsingTests/SourceItemTests.cs#L45) in SourceItemTests using [DeepEquals](https://github.com/NuGet/NuGet.Client/blob/dev/test/NuGet.Core.Tests/NuGet.Configuration.Test/SettingsTestUtils.cs#L81). Seems this method is assumed to compare if two sourceItems are exactly the same, including key, value, protocolVersion, allowInsecureConnects.
This DeepEquals will call into this Equals method.
With this change, the tests will not be able to compare every attributes.

Since I'm working on adding `allowInsecureConnects` to SourceItem, I'll change tests to validate the other properties when using DeepEquals.

Reply:
The code deleted only compares `Key` (source name) and `ProtocolVersion`, it never compared `Value` (source URL). So, `DeepEquals` would have issues anyway if it depends on `SourceItems.Equals`. In fact, the same is true for `AddItem`, so anything that uses it, like `<fallbackFolders>` and `<disabledPackageSources>`, has the same risk.